src/test/java/com/google/jenkins/plugins/computeengine/integration/ComputeEngineCloudSnapshotCreatedIT.java

```diff
@@ -121,16 +117,19 @@ public static void teardown() throws IOException {
     }
 
     /** Tests snapshot is created when we have failure builds for given node */
+    @WithTimeout(1200)
     @Test
     public void testSnapshotCreatedNotNull() {
         assertNotNull(createdSnapshot);
     }
 
+    @WithTimeout(1200)
     @Test
     public void testSnapshotCreatedStatusReady() {
         assertEquals("READY", createdSnapshot.getStatus());
     }
 
+    @WithTimeout(1200)
```

Comment:
Actually all the 3 tests here rely on the same `@ClassRule  init()`  which is doing the heavy lifting on provisioning the agent.

These tests are simply asserting the various state of the same agent.

So imo, we should simply put all of these assertions in a single test method; these 3 test methods seem to not solving specific issue.

Reply:
So we should simply put the `WithTimeout` at a single test, and just that..

Reply:
fixing it too...

Reply:
for now we will keep the `@ClassRule` based `JenkinsRule` just to not make far too many diff in single PR